File: official/docs/csharp/current/orders/one-call-buy.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;
using EasyPost.Parameters;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            Parameters.Order.Create parameters = new()
            {
                ToAddress = new Parameters.Address.Create
                {
                    Name = "Dr. Steve Brule",
                    Street1 = "417 Montgomery Street",
                    Street2 = "5th Floor",
                    City = "San Francisco",
                    State = "CA",
                    Country = "US",
                    Zip = "94104"
                },
                FromAddress = new Parameters.Address.Create
                {
                    Company = "EasyPost",
                    Street1 = "417 Montgomery Street",
                    Street2 = "Floor 5",
                    City = "San Francisco",
                    State = "CA",
                    Country = "US",
                    Zip = "94104"
                },
                Shipments = new List<IShipmentParameter>
                {
                    new Parameters.Shipment.Create()
                    {
                        Parcel = new Parameters.Parcel.Create
                        {
                            Weight = 10.2
                        }
                    },
                    new Parameters.Shipment.Create()
                    {
                        Parcel = new Parameters.Parcel.Create
                        {
                            PredefinedPackage = "FedExBox",
                            Weight = 17.5
                        }
                    }
                },
                Service = "NextDayAir",
                CarrierAccounts = new List<CarrierAccount>
                {
                    new CarrierAccount
                    {
                        Id = "ca_..."
                    }
                }
            };

            Order order = await client.Order.Create(parameters);

            Console.WriteLine(JsonConvert.SerializeObject(order, Formatting.Indented));
        }
    }
}

File: official/docs/golang/current/orders/one-call-buy.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v4"
)

func main() {
	client := easypost.New("EASYPOST_API_KEY")

	toAddress, _ := client.GetAddress("adr_...")
	fromAddress, _ := client.GetAddress("adr_...")

	firstShipment := &easypost.Shipment{
		Parcel: &easypost.Parcel{
			PredefinedPackage: "FedExBox",
			Weight:            10.4,
		},
	}

	secondShipment := &easypost.Shipment{
		Parcel: &easypost.Parcel{
			PredefinedPackage: "FedExBox",
			Weight:            17.5,
		},
	}

	order, _ := client.CreateOrder(
		&easypost.Order{
			Service:     "NextDayAir",
			ToAddress:   toAddress,
			FromAddress: fromAddress,
			Shipments: []*easypost.Shipment{
				firstShipment,
				secondShipment,
			},
		},
		&easypost.CarrierAccount{
			ID: "ca_...",
		},
	)

	fmt.Println(order)
}
